sql/unireg.cc

```diff
@@ -924,7 +924,6 @@ static bool pack_fields(File file, List<Create_field> &create_fields,
     recpos= field->offset+1 + (uint) data_offset;
     int3store(buff+5,recpos);
     int2store(buff+8,field->pack_flag);
-    DBUG_ASSERT(field->unireg_check < 256);
```

Comment:
IMHO what this is trying to say is compile_time_assert(sizeof() == 1)

Reply:
Yes, that would be the intention, but that check would fail - because the size of that field is 4 byte, not 1 - it just never contains a value higher than 256.